services/src/main/java/io/druid/cli/CliPeon.java

```diff
@@ -223,6 +224,15 @@ public void run()
             )
         );
         injector.getInstance(ExecutorLifecycle.class).join();
+
+        // Sanity check to help debug unexpected non-daemon threads
+        final Set<Thread> threadSet = Thread.getAllStackTraces().keySet();
+        for (Thread thread : threadSet) {
+          if (!thread.isDaemon() && thread != Thread.currentThread()) {
+            log.error("Thread [%s] is non daemon and would prevent shutdown! ", thread);
```

Comment:
This looks really scary for users but is not in fact actually reporting on a thing that should scare users. It's more for developers, right? So it should be info or even debug and worded a little less scarily.